libnetwork/ipams/defaultipam/structures.go

```diff
@@ -59,13 +68,44 @@ func PoolIDFromString(str string) (pID PoolID, err error) {
 	return pID, nil
 }
 
+func parsePoolIDV2(str string) (pID PoolID, err error) {
+	str = strings.TrimSuffix(strings.TrimPrefix(str, "PoolID{"), "}")
+
+	for _, field := range strings.FieldsFunc(str, func(c rune) bool { return c == ';' }) {
+		p := strings.SplitN(field, "=", 2)
```

Comment:
`strings.Cut` is a better option overall; https://pkg.go.dev/strings#Cut, but it would ignore multiple `=` in the string etc, in case we need to care about those.